### DISCUSSION TITLE
the problem with the duration of training

### DISCUSSION THREAD

**Discussion by KonradMaciaga** [trackmania-rl/tmrl#25] [Q&A] [answered]
hey, I wanted to use this project for my undergraduate thesis at university, but I encountered a problem. Every time I let go of training myself (whether with regular LIDAR or LIDAR progress with REDQSAC) the drive attempt ends up with a timeout after 50s of driving. Is the reason for this due to the limitations of my computer or have I not read up and the 50s of training is hardcoded in TMRL. Thanks in advance for the answer :D

**Comment by yannbouteiller** [comment, marked as answer]
Hello!

This is not really a timeout, this is the end (truncation) of an episode.

If you read the `config.json` file, you will find that a maximum of 1000 time steps per episode is defined there (you can change it if you like). You will also find that the time-step duration is set to 0.05 seconds by default (i.e., 20 FPS). The 1000 time-step limit multiplied by 0.05s is your 50s :)

This limit is there because the rollout worker waits until the end of the episode to send its gathered training samples to the server, so you don't want your episode to be infinitely long.

This is a design choice though, it is easy to adapt the code if people do want to play infinite episodes I believe.

**Reply by KonradMaciaga** [reply]
I update training_steps_per_round to 5000 but I still get an error after 50 seconds: It looks like this 
![Zrzut ekranu 2022-11-23 140001](https://user-images.githubusercontent.com/63426422/203553388-b29dfad6-b85b-4d20-abbd-139df434c9f9.png)

**Reply by yannbouteiller** [reply]
Yes training_steps_per_round is something else, it is just the number of training steps performed before the Trainer prints new results in the terminal. What you are looking for is `"RW_MAX_SAMPLES_PER_EPISODE"`
